src/main/java/com/google/devtools/build/lib/analysis/BuildView.java

```diff
@@ -419,6 +419,40 @@ public AnalysisResult update(
       skyframeBuildView.clearInvalidatedConfiguredTargets();
     }
 
+    ImmutableSet.Builder<ConfiguredTarget> incompatibleTargets = ImmutableSet.builder();
+    ImmutableSet.Builder<ConfiguredTarget> incompatibleButRequestedTargets = ImmutableSet.builder();
+
+    for (ConfiguredTarget target : skyframeAnalysisResult.getConfiguredTargets()) {
```

Comment:
In the interest in having clear "owners" for each feature concern (i.e. all logic addressing the same concern consolidated in the same place), is it possible to move most of this block into a static method on TopLevelConstraintSemantics?

If that ends up being hard because of unresolved dependencies on SkyframeAnalysisResult and such things, then alternatively a distinct method here in BuildView?

Reply:
Putting this in `TopLevelConstraintSemantics` sounds like a great idea. I've been slacking on cleaning this up.